tests/sample/reflect_packet.c

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+#include "bpf_helpers.h"
+#include "ebpf.h"
+#include "xdp_tests_common.h"
+
+void
+copy_mac_address(uint8_t destination[6], uint8_t source[6])
+{
+    for (int i = 0; i < 6; i++)
+        destination[i] = source[i];
+}
+
+void
+swap_mac_addresses(ETHERNET_HEADER* eth_hdr)
+{
+    uint8_t temp[6] = {0};
+    copy_mac_address(temp, eth_hdr->Destination);
+    copy_mac_address(eth_hdr->Destination, eth_hdr->Source);
+    copy_mac_address(eth_hdr->Source, temp);
+}
+
+void
+swap_ip_addresses(IPV4_HEADER* iphdr)
```

Comment:
don't use a version agnostic function name with a version specific type.
Samples need to support both IPv4 and IPv6.